### PR TITLE
drivers: sensor: Add OPT3001 light sensor driver

### PR DESCRIPTION
**Pull request by iosif-macesanu**
This pull request adds support for the Texas Instruments OPT3001 light sensor

### CODE REVIEW

**Comment by iosifmacesanu** [comment]
This driver should be compatible with the OPT3002 light sensor as well but for the sake of clarity we named it OPT3001 instead of OPT300x. Any objections?

File: drivers/sensor/opt3001/opt3001.c [hunk 1]
```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2019 Actinius
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <i2c.h>
+#include <sensor.h>
```
Comment (MaureenHelm) [inline drivers/sensor/opt3001/opt3001.c]: These should be:
```c
#include <drivers/i2c.h>
#include <drivers/sensor.h>
```

File: drivers/sensor/opt3001/opt3001.c [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+			.len = 2,
+			.flags = I2C_MSG_READ,
+		},
+	};
+
+	if (send_stop) {
+		msgs[1].flags |= I2C_MSG_STOP;
+	}
+
+	if (i2c_transfer(drv_data->i2c, msgs,
```
Comment (MaureenHelm) [inline drivers/sensor/opt3001/opt3001.c]: `i2c_burst_read` would be better here, as you would not have to manually construct the msgs array

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/opt3001/opt3001.c (hunk 1 above)
Comment (iosifmacesanu) [reply drivers/sensor/opt3001/opt3001.c]: Updated.

File: drivers/sensor/opt3001/opt3001.c (hunk 2 above)
Comment (iosifmacesanu) [reply drivers/sensor/opt3001/opt3001.c]: Indeed. Updated.

**Comment by iosifmacesanu** [comment]
Thanks for your suggestions @MaureenHelm

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**